core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -49,6 +54,7 @@ object MultiCloudWorkspaceService {
                   billingProfileManagerDAO: BillingProfileManagerDAO,
                   samDAO: SamDAO,
                   multiCloudWorkspaceConfig: MultiCloudWorkspaceConfig,
+                  leonardoConfig: LeonardoConfig,
                   workbenchMetricBaseName: String
   )(ctx: RawlsRequestContext)(implicit ec: ExecutionContext, system: ActorSystem): MultiCloudWorkspaceService =
     new MultiCloudWorkspaceService(
```

Comment:
Here's some structural/architectural suggestions:
* instead of accepting `leonardoConfig: LeonardoConfig` as an argument, accept a `leonardoDAO: LeonardoDAO`. Construct the DAO in `Boot` to supply to this argument. This would allow you, in unit tests, to supply a mock dao and still test this service.
* move the `appType = "CROMWELL"` value out of the `multiCloudWorkspaces.azureConfig` config block and into the `leonardo` config block, and make it part of the `LeonardoConfig` class. This eliminates a bunch of changes in `MultiCloudWorkspaceConfig`; it also feels like it encapsulates Leo app names inside Leo-related classes instead of in Azure-wide classes. The only classes that would be looking at that config value are now `LeonardoConfig` and `HttpLeonardoDAO`. Might also want to rename `appType = "CROMWELL"` to `wdsType = "CROMWELL"` to make it clear which "app" you're concerned with.
* here in `MultiCloudWorkspaceService` - assuming you've broken out a generic `createApp()` and a WDS-specific `createWdsApp()` in `LeonardoDAO` - you can call the more specific `createWdsApp()`. This further encapsulates logic into the Leo dao and out of this class.